common/src/api/internal/shared.rs

```diff
@@ -1101,6 +1101,50 @@ pub struct SledIdentifiers {
     pub serial: String,
 }
 
+/// Identifies the kind of CPU present on a sled, determined by reading CPUID.
+///
+/// This is intended to broadly support the control plane answering the question
+/// "can I run this instance on that sled?" given an instance with either no or
+/// some CPU platform requirement. It is not enough information for more precise
+/// placement questions - for example, is a CPU a high-frequency part or
+/// many-core part? We don't include Genoa here, but in that CPU family there
+/// are high frequency parts, many-core parts, and large-cache parts. To support
+/// those questions (or satisfactorily answer #8730) we would need to collect
+/// additional information and send it along.
+#[derive(
+    Serialize, Deserialize, Copy, Clone, Debug, PartialEq, Eq, JsonSchema,
+)]
+#[serde(rename_all = "snake_case")]
+pub enum SledCpuFamily {
```

Comment:
Nit, take it or leave it: is `omicron-common` really the best place for this, or should it just go in `nexus-sled-agent-shared`? More crates depend on `omicron-common`, and AFAICT, this is only needed by Nexus and sled-agent. I don't think it's _super_ important to keep it out of `omicron-common` for compile-time reasons (it's just one enum, how much could it cost?), but it kinda seems like a better habit to put stuff that's only shared between Nexus and sled-agent in the crate that's for that?

Reply:
okay, settled on this going (back!) to `sled-hardware`. the initial version here had a `sled_hardware::SledCpuFamily`, a copy with the same variants and more docs in `nexus_types`, and then the database enum in `nexus_db_model`.

I think the intention was that sled-agent's internal representation (for the fleeting time that we're in `detect_cpu_family()`) shouldn't be coupled to whatever we've got in the API, but I don't think we'll ever actually _differ_ in those. we're dancing around the kernel's `x86_processor_family_t` here, where in the ideal world we're just scooping that out of `libtopo` and passing that to Nexus. so I don't feel too bad about having the same type all through here for internal purposes.